src/main/java/com/dragonballs/security/WebSecurityConfig.java

```diff
@@ -0,0 +1,43 @@
+package com.dragonballs.security;
+
+import com.dragonballs.services.authentication.JwtAuthorizationTokenFilter;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.security.config.annotation.web.builders.HttpSecurity;
+import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
+import org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter;
+import org.springframework.security.config.http.SessionCreationPolicy;
+import org.springframework.security.web.authentication.UsernamePasswordAuthenticationFilter;
+
+@EnableWebSecurity
+@Configuration
+public class WebSecurityConfig extends WebSecurityConfigurerAdapter {
+
+    @Autowired
+    JwtAuthorizationTokenFilter authenticationTokenFilter;
+
+    @Override
+    protected void configure(HttpSecurity httpSecurity) throws Exception {
+        //http.csrf().disable();
+        httpSecurity
+                // we don't need CSRF because our token is invulnerable
+                .csrf().disable()
```

Comment:
Does it mean that every time user would open the page, he'd be forced to re-enter his credentials again? https://security.stackexchange.com/a/166798

Reply:
Keeping session is done in frontend part. We can always check if user token is valid refresh it so it won't timeout, assuming user takes action. As we have discussed when we were developing backend part of login.

Reply:
Totally agree, but the only way to do that is to store it in cookies - in that case, as per info above, CSRF is needed.